### PR TITLE
Scale histograms to data and aggregate groups in histmaker postprocessing steps

### PR DESCRIPTION
**Pull request by davidwalter2**
See title. The new functionalities are enabled by default and can be disabled by command line arguments.

Two other small changes:
- Propagate argument to forceNonzero to be used in top level scripts.
- Add argument to check systematics in cardmaker

### CODE REVIEW

**Comment by davidwalter2** [comment]
This should be ready for review

File: scripts/histmakers/mw_with_mu_eta_pt.py [hunk 1]
```diff
@@ -653,7 +657,13 @@ def build_graph(df, dataset):
     return results, weightsum
 
 resultdict = narf.build_and_run(datasets, build_graph)
+
+if not args.noScaleToData:
+    scale_to_data(resultdict)
+    aggregate_groups(datasets, resultdict, groups_to_aggregate)
+
 if not args.onlyMainHistograms and args.muonScaleVariation == 'smearingWeights':
+    logger.debug("Apply smaringWeights")
```
Comment (kdlong) [inline scripts/histmakers/mw_with_mu_eta_pt.py:666]: typo ;)

Comment (davidwalter2) [reply scripts/histmakers/mw_with_mu_eta_pt.py]: done

**Comment by davidwalter2** [comment]
Compared to #236 (latest merged PR) and #233 (latest PR that changed the result) plots look almost identical. A very little change is visible that may be just a rounding effect or also related to `forceNonZero` which is with this PR only applied after summing up certain groups (before it was done before). Was not able to trace back the exact cause but it's negligible.
Using --aggregateGroups --noScaleToData restores the current procedure exactly, this I verified in a test. 

I would merge this PR soon if no one has any objection

**Pull request merged**